table/block.cc

```diff
@@ -283,9 +291,33 @@ bool BlockIter::ParseNextKey() {
 
     value_ = Slice(p + non_shared, value_length);
     while (restart_index_ + 1 < num_restarts_ &&
-           GetRestartPoint(restart_index_ + 1) < current_) {
+           GetRestartPoint(restart_index_ + 1) <= current_) {
```

Comment:
This was a bug in the existing code. The bug did not do any harm though and the restart point would be eventually be set correctly.

Reply:
I read the code a little bit. Maybe I'm wrong but before your new code, `restart_index_` seems to only be useful when calling Prev(). In that case, in that case `GetRestartPoint(restart_index_) == current_` is not very useful, as the current entry is not interesting, so we are going `restart_index_--` for that anyway, so we are doing an unnecessary ++ and --, as well as an extra `GetRestartPoint()` in Prev().

Reply:
First I would say that existing code is confusing as restart_index_ does not reflect what its definition says, and would make it likely to create bugs. But if the performance overhead of the one more GetRestartPoint is the concern I can revert it back to previous implementation and does the additional call only if value_delta_encoded_ is enabled.

Reply:
update: this is reverted now.